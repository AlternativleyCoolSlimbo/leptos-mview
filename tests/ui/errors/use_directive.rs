use leptos::{html::AnyElement, HtmlElement};
use leptos_mview::mview;

fn no_arg_dir(_el: HtmlElement<AnyElement>) {}

fn arg_dir(_el: HtmlElement<AnyElement>, _argument: i32) {}

fn missing_argument() {
    _ = mview! {
        div use:arg_dir;
    };
}

fn extra_argument() {
    _ = mview! {
        span use:no_arg_dir=2;
    };
}

fn main() {}
