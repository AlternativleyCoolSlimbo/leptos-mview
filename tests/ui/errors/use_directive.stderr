error[E0277]: the trait bound `i32: From<()>` is not satisfied
  --> tests/ui/errors/use_directive.rs:10:17
   |
10 |         div use:arg_dir;
   |                 ^^^^^^^ the trait `From<()>` is not implemented for `i32`, which is required by `(): Into<_>`
   |
   = help: the following other types implement trait `From<T>`:
             <i32 as From<bool>>
             <i32 as From<i16>>
             <i32 as From<i8>>
             <i32 as From<u16>>
             <i32 as From<u8>>
   = note: required for `()` to implement `Into<i32>`

error[E0277]: the trait bound `(): From<{integer}>` is not satisfied
  --> tests/ui/errors/use_directive.rs:16:29
   |
15 |       _ = mview! {
   |  _________-
16 | |         span use:no_arg_dir=2;
   | |                             ^ the trait `From<{integer}>` is not implemented for `()`, which is required by `{integer}: Into<_>`
17 | |     };
   | |_____- required by a bound introduced by this call
   |
   = help: the following other types implement trait `From<T>`:
             <(T, T) as From<[T; 2]>>
             <(T, T, T) as From<[T; 3]>>
             <(T, T, T, T) as From<[T; 4]>>
             <(T, T, T, T, T) as From<[T; 5]>>
             <(T, T, T, T, T, T) as From<[T; 6]>>
             <(T, T, T, T, T, T, T) as From<[T; 7]>>
             <(T, T, T, T, T, T, T, T) as From<[T; 8]>>
             <(T, T, T, T, T, T, T, T, T) as From<[T; 9]>>
           and $N others
   = note: required for `{integer}` to implement `Into<()>`
